### DISCUSSION TITLE
when +server.js throws an error, it shows error page that I did not create

### DISCUSSION THREAD

**Discussion by Madd0g** [sveltejs/kit#7234] [Q&A] [answered]
I have copied the random number example from https://kit.svelte.dev/docs/routing#server

My file structure is:

```
/routes/+error.svelte
/routes/random/+server.js
```

When the error is triggered, I see this error page, but this is not what my `+error.svelte` looks like and it doesn't have my layout. When I trigger an error in regular svelte components, I see my `+error.svelte` as expected.

<img width="753" alt="image" src="https://user-images.githubusercontent.com/1171003/195171882-1d5364b8-1564-4e10-a702-cc3700f748ad.png">

Do you know why this happens, and how I can customize the error response?

**Comment by teemingc** [comment, marked as answer]
`+error.svelte` is only displayed when an error occurs in a `load` function
>If an error occurs during load, SvelteKit will render a default error page. You can customise this error page on a per-route basis by adding an `+error.svelte` file:

You can customise error responses returned from endpoints by customising a `src/error.html` file.
> SvelteKit will bail out and render a static fallback error page, which you can customise by creating a `src/error.html` file.

https://kit.svelte.dev/docs/routing#error

Hope this helps! (I have not tested these. Only assuming based on the docs).

**Reply by Madd0g** [reply]
Huh the docs make it sound like `error.html` is a fallback for `+error.svelte`, which is probably why I missed it. I'll test it.

Kinda strange to have a separate way for endpoints.

**Reply by teemingc** [reply]
> Huh the docs make it sound like `error.html` is a fallback for `+error.svelte`

I think it also is a fallback for it.

A probable summary:
1. `+error.svelte` when pages fail to load
2. `error.html` when `+error.svelte` or `+server` fails.

**Reply by dummdidumm** [reply]
If you visit the endpoint directly, the request headers default to html, so the server responds with the `error.html`. If you fetch this from JS, that request header is different so it returns JSON in that case.

**Reply by Madd0g** [reply]
thanks, makes sense
